crates/rust-analyzer/src/reload.rs

```diff
@@ -453,19 +445,29 @@ impl GlobalState {
     }
 
     fn fetch_build_data_error(&self) -> Option<String> {
-        let mut buf = String::new();
+        let mut buf = "rust-analyzer failed to run build scripts:\n".to_string();
+        let mut has_errors = false;
 
         for ws in &self.fetch_build_data_queue.last_op_result().1 {
-            if let Err(err) = ws {
-                stdx::format_to!(buf, "rust-analyzer failed to run custom build: {:#}\n", err);
+            match ws {
+                Ok(data) => {
+                    if let Some(err) = data.error() {
+                        has_errors = true;
+                        stdx::format_to!(buf, "{:#}\n", err);
+                    }
```

Comment:
This was semi-intentional -- if `cargo check` fails because you have a type error in non-build.rs code, we should consider that as a working workspace. I *think* this is a non-issue given that we use RUSTC_WRAPPER trick, but it makes sense to re-check that you don't get partially loaded workspace notification

![image](https://user-images.githubusercontent.com/1711539/130825459-7eea70d9-a4dc-467c-9d85-c00ee1586f44.png)


just becaues borrow checked doesn't favor you today.